src/Integration/LocalServices/ISourceControlledFileSystem.cs

```diff
@@ -35,11 +36,21 @@ internal interface ISourceControlledFileSystem : ILocalService
         /// <param name="fileWriteOperation">The actual write operation, return false if failed</param>
         void QueueFileWrite(string filePath, Func<bool> fileWriteOperation);
 
+        /// <summary>
+        /// Queues a write operation for given file paths. New or edit is determined internally at the time of this method execution.
+        /// </summary>
+        void QueueFileWrites(IList<string> filePaths, Func<bool> fileWriteOperation);
```

Comment:
Miscellaneous note: we should rename this interface at some point - it's misleading to call it an abstraction over the file system. It would more accurate to call it a "SourceControlOperationQueue"